src/redux/selectors/index.js

```diff
@@ -196,16 +178,140 @@ export const patientAgeAtResourcesSelector = createSelector(
       return {};
     }
     const birthDate = parse(patient?.birthDate, 'yyyy-MM-dd', new Date());
-    return timelineItems.reduce((acc, { id, timelineDate }) => {
+    const patientAgeAtResources = {};
+    timelineItems.forEach(({ id, timelineDate }) => {
       const resourceDate = format(new Date(timelineDate), 'yyyy-MM-dd');
       const ageAtResourceDate = intervalToDuration({
         start: birthDate,
         end: parse(resourceDate, 'yyyy-MM-dd', new Date()),
       });
+      if (!patientAgeAtResources[id]) {
+        patientAgeAtResources[id] = ageAtResourceDate;
+      }
+    });
+    return patientAgeAtResources;
+  },
+);
+
+export const lastAddedResourceIdSelector = createSelector(
+  [collectionsSelector, selectedCollectionSelector],
+  (collections, selectedCollectionId) => collections[selectedCollectionId].lastAddedResourceId,
+);
+
+export const collectionResourceIdsSelector = createSelector(
+  [collectionsSelector, selectedCollectionSelector],
+  (collections, selectedCollectionId) => collections[selectedCollectionId].resourceIds,
+);
+
+const selectedCollectionResourceIdsSelector = createSelector(
+  [collectionsSelector, selectedCollectionSelector],
+  (collections, selectedCollection) => collections[selectedCollection]?.resourceIds,
+);
 
-      acc[id] = ageAtResourceDate;
+const isResourceInDateRange = (resource, dateRangeStart, dateRangeEnd) => {
+  const { timelineDate } = resource;
+  if (!timelineDate) {
+    console.info('record does not have an timelineDate, resourceId: ', resource.id); // eslint-disable-line no-console
+    return false;
+  }
+  if (!dateRangeStart || !dateRangeEnd) { return true; }
+  return (
+    isWithinInterval(
+      timelineDate, { start: startOfDay(dateRangeStart), end: endOfDay(dateRangeEnd) },
+    )
+  );
+};
+
+const filteredResourceTypesSelector = createSelector(
+  [
+    resourceTypeFiltersSelector,
+    resourceIdsGroupedByTypeSelector,
+    selectedResourceTypeSelector,
+    selectedCollectionResourceIdsSelector,
+    dateRangeFilterFiltersSelector,
+    resourcesSelector,
+    timelinePropsSelector,
+  ],
+  (
+    resourceTypeFilter,
+    resourceIdsGroupedByType,
+    selectedResourceType,
+    selectedCollectionResourceIdsObjects,
+    dateRangeFilterFilters,
+    resources,
+    timelineProps,
+  ) => {
+    if (!selectedCollectionResourceIdsObjects) {
+      return {};
+    }
+    const { minimumDate, maximumDate } = timelineProps;
+    const { dateRangeStart, dateRangeEnd } = dateRangeFilterFilters;
+    const activeDateStart = dateRangeStart || minimumDate;
+    const activeDateEnd = dateRangeEnd || maximumDate;
```

Comment:
```
const { minimumDate, maximumDate } = timelineProps;
const { dateRangeStart = minimumDate, dateRangeEnd = maximumDate } = dateRangeFilterFilters;
```
and dispense w/ `activeDateStart` / `activeDateEnd` ?

Reply:
there is also `timelineRangeSelector`, that does ^this.

Reply:
or use `timelineItemsInRangeSelector` (see comment below) and modify `pickTimelineFields` to also pluck `subType`)  -- eliminating any other time-related selector fed to this one.

Reply:
This will have to be a refactor after the demo

Reply:
`timelineItemsInRangeSelector` is maybe the safest option. it tests whether there _is_ a start/end date, and otherwise returns an empty Array.

...This is an important check, b/c it is conceivable that particular UI views render before any records with dates are processed, and that would prevent an exception, here.

Reply:
could derive a selector that combines type and date filters.   (probably more performant: type 1st, range testing, 2nd)